java/query/query-core/src/main/java/sleeper/query/core/recordretrieval/QueryExecutor.java

```diff
@@ -142,6 +169,14 @@ public CloseableIterator<Row> execute(Query query) throws QueryException {
         return new ConcatenatingIterator(iteratorSuppliers);
     }
 
+    /**
+     * Executes the query and returns records. The records are not returned in any
+     * particular order.
```

Comment:
It seems important to note that this is used internally by Sleeper, rather than by a client. This takes a sub-query, which has been enriched by Sleeper with information about a single partition. That should generally only be done internally.